### ISSUE TITLE
Data extraction using bounds raises TypeError

### ISSUE THREAD

**Issue by rdenham** [jgrss/geowombat#22]
When trying to extract data using a tuple of bounded coordinates, I get an error. Here is the example taken from the documentation:
```
bounds = (793000.0, 2049000.0, 794000.0, 2050000.0)

with gw.open(rgbn,
             band_names=['green', 'red', 'nir'],
             num_workers=8,
             indexes=[2, 3, 4],
             bounds=bounds,
             out_dtype='float32') as ds:
    print(ds)
```

And the error:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/anaconda3/envs/gwenv/lib/python3.7/site-packages/numpy/core/function_base.py in linspace(start, stop, num, endpoint, retstep, dtype, axis)
    116     try:
--> 117         num = operator.index(num)
    118     except TypeError:

TypeError: 'float' object cannot be interpreted as an integer

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-18-26e292b10c6b> in <module>
      6              indexes=[2, 3, 4],
      7              bounds=bounds,
----> 8              out_dtype='float32') as ds:
      9     print(ds)

~/anaconda3/envs/gwenv/lib/python3.7/contextlib.py in __enter__(self)
    110         del self.args, self.kwds, self.func
    111         try:
--> 112             return next(self.gen)
    113         except StopIteration:
    114             raise RuntimeError("generator didn't yield") from None

~/anaconda3/envs/gwenv/lib/python3.7/site-packages/geowombat/core/api.py in open(filename, return_as, band_names, time_names, stack_dim, bounds, bounds_by, resampling, mosaic, overlap, dtype, num_workers, **kwargs)
    348                     bounds=bounds,
    349                     num_workers=num_workers,
--> 350                     **kwargs)
    351 
    352     else:

~/anaconda3/envs/gwenv/lib/python3.7/site-packages/geowombat/core/api.py in read(filename, band_names, time_names, bounds, num_workers, **kwargs)
    162                 kwargs['window'] = from_bounds(*bounds, transform=src.transform)
    163 
--> 164             ycoords, xcoords, attrs = get_attrs(src, **kwargs)
    165 
    166         data = dask.compute(read_delayed(filename, 0, **kwargs),

~/anaconda3/envs/gwenv/lib/python3.7/site-packages/geowombat/core/api.py in get_attrs(src, **kwargs)
     35     ycoords = np.linspace(src.bounds.top - (kwargs['window'].row_off * src.res[0]),
     36                           src.bounds.top - (kwargs['window'].row_off * src.res[0]) -
---> 37                           (kwargs['window'].height * src.res[0]), kwargs['window'].height)
     38 
     39     xcoords = np.linspace(src.bounds.left + (kwargs['window'].col_off * src.res[0]),

<__array_function__ internals> in linspace(*args, **kwargs)

~/anaconda3/envs/gwenv/lib/python3.7/site-packages/numpy/core/function_base.py in linspace(start, stop, num, endpoint, retstep, dtype, axis)
    119         raise TypeError(
    120             "object of type {} cannot be safely interpreted as an integer."
--> 121                 .format(type(num)))
    122 
    123     if num < 0:

TypeError: object of type <class 'float'> cannot be safely interpreted as an integer.
```

This is caused in the call to `numpy.linspace` using the calculated window width as the argument `num`. But the window width as calculated by `rio.window.from_bounds` returns a float not an integer.

Perhaps this should be cast to an integer in the `linspace` call?

My pip freeze output:
```
affine==2.3.0
alabaster==0.7.12
asciitree==0.3.3
attrs==19.3.0
Babel==2.8.0
backcall==0.1.0
bleach==3.1.0
certifi==2019.11.28
cftime==1.0.4.2
chardet==3.0.4
Click==7.0
click-plugins==1.1.1
cligj==0.5.0
cloudpickle==1.2.2
cycler==0.10.0
Cython==0.29.14
dask==2.9.2
dask-glm==0.2.0
dask-ml==1.2.0
dateparser==0.7.2
decorator==4.4.1
defusedxml==0.6.0
Deprecated==1.2.7
distributed==2.9.3
docutils==0.16
entrypoints==0.3
fasteners==0.15
Fiona==1.8.13
fsspec==0.6.2
GDAL==2.4.0
geopandas==0.6.2
geowombat==1.0.7
h5netcdf==0.7.4
h5py==2.10.0
HeapDict==1.0.1
idna==2.8
imagesize==1.2.0
importlib-metadata==1.4.0
ipykernel==5.1.3
ipython==7.11.1
ipython-genutils==0.2.0
jedi==0.16.0
Jinja2==2.10.3
joblib==0.14.1
json5==0.8.5
jsonschema==3.2.0
jupyter-client==5.3.4
jupyter-core==4.6.1
jupyterlab==1.2.6
jupyterlab-server==1.0.6
kiwisolver==1.1.0
latexcodec==2.0.0
llvmlite==0.31.0
locket==0.2.0
MarkupSafe==1.1.1
matplotlib==3.1.2
mistune==0.8.4
mkl-fft==1.0.15
mkl-random==1.1.0
mkl-service==2.3.0
monotonic==1.5
msgpack==0.6.2
multipledispatch==0.6.0
munch==2.5.0
nbconvert==5.6.1
nbformat==5.0.4
netCDF4==1.5.3
notebook==6.0.3
numba==0.47.0
numcodecs==0.6.4
numexpr==2.7.1
numpy==1.18.1
numpydoc==0.9.2
opencv-python==4.1.2.30
oset==0.1.3
packaging==20.1
pandas==0.25.3
pandocfilters==1.4.2
parso==0.5.2
partd==1.1.0
pexpect==4.8.0
pickleshare==0.7.5
prometheus-client==0.7.1
prompt-toolkit==3.0.2
psutil==5.6.7
ptyprocess==0.6.0
pybtex==0.22.2
pybtex-docutils==0.2.2
Pygments==2.5.2
pyparsing==2.4.6
pyproj==2.4.2.post1
pyrsistent==0.15.7
python-dateutil==2.8.1
pytz==2019.3
PyYAML==5.3
pyzmq==18.1.1
rasterio==1.1.2
regex==2020.1.8
requests==2.22.0
scikit-learn==0.22.1
scipy==1.4.1
Send2Trash==1.5.0
Shapely==1.6.4.post2
six==1.14.0
snowballstemmer==2.0.0
snuggs==1.4.7
sortedcontainers==2.1.0
Sphinx==2.3.1
sphinx-automodapi==0.12
sphinxcontrib-applehelp==1.0.1
sphinxcontrib-bibtex==1.0.0
sphinxcontrib-devhelp==1.0.1
sphinxcontrib-htmlhelp==1.0.2
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.2
sphinxcontrib-serializinghtml==1.1.3
tblib==1.6.0
terminado==0.8.3
testpath==0.4.4
toolz==0.10.0
tornado==6.0.3
tqdm==4.42.0
traitlets==4.3.3
tzlocal==2.0.0
urllib3==1.25.8
wcwidth==0.1.8
webencodings==0.5.1
wget==3.2
wrapt==1.11.2
xarray==0.14.1
zarr==2.4.0
zict==1.0.0
zipp==2.1.0affine==2.3.0
alabaster==0.7.12
asciitree==0.3.3
attrs==19.3.0
Babel==2.8.0
backcall==0.1.0
bleach==3.1.0
certifi==2019.11.28
cftime==1.0.4.2
chardet==3.0.4
Click==7.0
click-plugins==1.1.1
cligj==0.5.0
cloudpickle==1.2.2
cycler==0.10.0
Cython==0.29.14
dask==2.9.2
dask-glm==0.2.0
dask-ml==1.2.0
dateparser==0.7.2
decorator==4.4.1
defusedxml==0.6.0
Deprecated==1.2.7
distributed==2.9.3
docutils==0.16
entrypoints==0.3
fasteners==0.15
Fiona==1.8.13
fsspec==0.6.2
GDAL==2.4.0
geopandas==0.6.2
geowombat==1.0.7
h5netcdf==0.7.4
h5py==2.10.0
HeapDict==1.0.1
idna==2.8
imagesize==1.2.0
importlib-metadata==1.4.0
ipykernel==5.1.3
ipython==7.11.1
ipython-genutils==0.2.0
jedi==0.16.0
Jinja2==2.10.3
joblib==0.14.1
json5==0.8.5
jsonschema==3.2.0
jupyter-client==5.3.4
jupyter-core==4.6.1
jupyterlab==1.2.6
jupyterlab-server==1.0.6
kiwisolver==1.1.0
latexcodec==2.0.0
llvmlite==0.31.0
locket==0.2.0
MarkupSafe==1.1.1
matplotlib==3.1.2
mistune==0.8.4
mkl-fft==1.0.15
mkl-random==1.1.0
mkl-service==2.3.0
monotonic==1.5
msgpack==0.6.2
multipledispatch==0.6.0
munch==2.5.0
nbconvert==5.6.1
nbformat==5.0.4
netCDF4==1.5.3
notebook==6.0.3
numba==0.47.0
numcodecs==0.6.4
numexpr==2.7.1
numpy==1.18.1
numpydoc==0.9.2
opencv-python==4.1.2.30
oset==0.1.3
packaging==20.1
pandas==0.25.3
pandocfilters==1.4.2
parso==0.5.2
partd==1.1.0
pexpect==4.8.0
pickleshare==0.7.5
prometheus-client==0.7.1
prompt-toolkit==3.0.2
psutil==5.6.7
ptyprocess==0.6.0
pybtex==0.22.2
pybtex-docutils==0.2.2
Pygments==2.5.2
pyparsing==2.4.6
pyproj==2.4.2.post1
pyrsistent==0.15.7
python-dateutil==2.8.1
pytz==2019.3
PyYAML==5.3
pyzmq==18.1.1
rasterio==1.1.2
regex==2020.1.8
requests==2.22.0
scikit-learn==0.22.1
scipy==1.4.1
Send2Trash==1.5.0
Shapely==1.6.4.post2
six==1.14.0
snowballstemmer==2.0.0
snuggs==1.4.7
sortedcontainers==2.1.0
Sphinx==2.3.1
sphinx-automodapi==0.12
sphinxcontrib-applehelp==1.0.1
sphinxcontrib-bibtex==1.0.0
sphinxcontrib-devhelp==1.0.1
sphinxcontrib-htmlhelp==1.0.2
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.2
sphinxcontrib-serializinghtml==1.1.3
tblib==1.6.0
terminado==0.8.3
testpath==0.4.4
toolz==0.10.0
tornado==6.0.3
tqdm==4.42.0
traitlets==4.3.3
tzlocal==2.0.0
urllib3==1.25.8
wcwidth==0.1.8
webencodings==0.5.1
wget==3.2
wrapt==1.11.2
xarray==0.14.1
zarr==2.4.0
zict==1.0.0
zipp==2.1.0
```

**Comment by jgrss** [comment]
Thanks for pointing this out. The `bounds` argument triggers a read using `dask.delayed`. I will look into this issue. In the meantime, you can achieve the same read by:

```
bounds = (793000.0, 2049000.0, 794000.0, 2050000.0)

with gw.config.update(ref_bounds=bounds):

    with gw.open(rgbn,
                          band_names=['green', 'red', 'nir'],
                          dtype='float32') as ds:
        print(ds)
```

**Comment by jgrss** [comment]
@rdenham There were multiple issues with `dask.delayed` reads. The first issue regarding coordinate creation was fixed in [7f5a03e431fda52b37a985026db85219407bf8b9](https://github.com/jgrss/geowombat/commit/7f5a03e431fda52b37a985026db85219407bf8b9). All fixes were pushed to `v1.1.1`.
